### ISSUE TITLE
Binding doesn't work in Sample Project's SampleDialog

### ISSUE THREAD

**Issue by vector-man** [derfsplat/MiniMVP#3]
In the example, your presenter is not being set in SampleDialog. I have been converting my project to MiniMVP, and I'm stuck trying to get my dialogs to get a Presenter, going by the sample project.  

SampleDialog:

        public partial class SampleDialog : Dialog<IPresentSampleDialog>, ISampleDialogView
        {
        public override void Bind(IPresentSampleDialog presenter)
        {
            base.Bind(presenter);
        }
        public SampleDialog()
        {
            InitializeComponent();

            Shown += SampleDialog_Shown;
        }
        private void SampleDialog_Shown(...)
        {
            // Presenter is null.
            Presenter.DoSomething()
        }
    ...
SampleDialogPresenter:

    public class SampleDialogPresenter : DialogPresenter<ISampleDialogView>, IPresentSampleDialog
    {
        private SimpleDataModel model;

        public SampleDialogPresenter(Func<ISampleDialogView> viewFactory)
            : base(viewFactory)
        {
        }

        #region IPresentSampleDialog Members

        //todo: have mainform implement IHostWinForms iface so we can get owner via IHOstWinForms dep passed to ctor
        public DialogResult ShowDialog(IWin32Window owner, SimpleDataModel model)
        {
            this.model = model;

            using (var view = viewFactory())
            {
                view.CanClose = () =>
                    {
                        if (string.IsNullOrWhiteSpace(view.TitleText))
                        {
                            MessageBox.Show("no no mr. smith no here.");
                            return false;
                        }

                        model.Title = view.TitleText;
                        return true;
                    };

                return view.ShowDialog(owner);
            }
        }

        public SimpleDataModel TitleModel
        {
            get
            {
                return model;
            }
        }
        public void DoSomething() { ... }
    ...

It looks like it is not automatic this way. 

Edit: It looks like the problem is that the way the demo is coded, the view's bind method is never called (even if overridden). This is because it is calling the view's ShowDialog() method instead of the Presenter's Show method.. I assume that ShowInternal() is also not called when I use ShowDialog() like above.

It seems I can do this, but I'm not sure if is the right way to use your library. Any thoughts on this method?:

     public DialogResult ShowDialog(IWin32Window owner, SimpleDataModel model)
     {
            this.model = model;
            using (var view = viewFactory())
            {
                View = view;
                View.Bind(this);
                view.CanClose = () =>
                    {
                        if (string.IsNullOrWhiteSpace(view.TitleText))
                        {
                            MessageBox.Show("no no mr. smith no here.");
                            return false;
                        }

                        model.Title = view.TitleText;
                        return true;
                    };
                return view.ShowDialog(owner);
            }
        }
 Do you know of a more elegant workaround to show a dialog with an owner or is this a decent way?

**Comment by derfsplat** [comment]
You are correct on all fronts- the challenge here is that Bind isn't properly supported when calling `ShowDialog` - in fact, as you stated, it bypasses the framework entirely.  

I used an approach like you have above in the project I wrote this for (calling `View.Bind(this);` myself), but let me think about this one a little more bc this is a terrible Api experience - less "pit of success", more "pit of despair" :)

**Comment by vector-man** [comment]
Thanks, I'm glad you are still working on it! It's a good framework. If I come up with any ideas, I'll share.

**Comment by derfsplat** [comment]
I cooked up an example over the weekend that removes all `Dialog` related classes and adds `ShowDialog` to `View`- i think this is more flexible in that you can show a window as an Mdi Child, Dialog, or non-modal window using the same base class, `View`, however it's less semantic.
I want to put together another example that makes proper use of the Dialog classes (e.g. `DialogPresenter`).  This will be more semantic, and perhaps clearer in intent at the Api level, but my concern is it'll be less flexible and/or more complex of an Api experience.

When I'm done with the 2nd example, I'll submit PR's for both and if you can spare the time I'd love your input :).

**Comment by vector-man** [comment]
Sure, no problem! Thanks!
